File: hotel/backend/src/main/kotlin/loshica/api/hotel/errors/EntityNotFoundError.kt
package loshica.api.hotel.errors

import loshica.api.hotel.core.BaseError
import org.springframework.http.HttpStatus

class EntityNotFoundError(private val entityName: String, private val id: Int) : BaseError(
    HttpStatus.NOT_FOUND,
    ApiError(
        code = HttpStatus.NOT_FOUND,
        message = "$entityName with id '$id' not found"
    )
)

File: computer-club/backend/src/shared/errors/entity-not-found.error.ts
import { BaseError } from './base.error';
import { HttpStatus } from '@nestjs/common';

export class EntityNotFoundError extends BaseError {
  constructor(entityName: string) {
    super(HttpStatus.NOT_FOUND, `Не удалось найти сущность '${entityName}'`);
  }
}
